raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1647,9 +1675,12 @@ private void handlePendingAppends(long currentTimeMs) {
      *
      * @param records The records to write to the log
      * @param ackMode The commit mode for the appended records
+     * @param timeoutMs The maximum time to wait for the append operation to complete (including
+     *                  any time needed for replication)
      * @return The uncommitted base offset and epoch of the appended records
      */
-    private CompletableFuture<OffsetAndEpoch> append(Records records, AckMode ackMode) {
+    @Override
+    public CompletableFuture<OffsetAndEpoch> append(Records records, AckMode ackMode, long timeoutMs) {
```

Comment:
While reading the implementation of `ReplicaCounter`, I realized that we are asserting an implicit assumption such that the listener registered on the future would be triggered in order --- they are, indeed today --- and if that's by-design we should make that clear in the docs and also stick with it to not make RaftClient be possibly multi-threaded out-of-ordering executing in the future.